Development into master

MelWttr opened the pull request.

Commit:
feat(fonts.scss): создает файл со шрифтами

Commit:
feat(css.reset.scss): создает файл reset

создает файл сброса стилей по-умолчанию

Commit:
build: добавляет конфиги для вебкака

Commit:
feat: добавляет верстку кнопок

добавлена верстка кнопок смены звуковых сигналов + стили

Commit:
feat: добавляет аудио и слайдер громкости

добавляет смену аудио и фона по нажатию кнопки
добавляет слайдер громкости

Commit:
build: добавляет typescript в проект

Commit:
feature(index.ts): добавляет типизацию в код

taraswww777 commented on public/index.html:
```diff
@@ -4,10 +4,49 @@
 	</head>
-	<body>
+	<body class="summer">
 		<main class="main">
```
### Форматирование кода

Код должен хоть как-то однообразно форматироваться, для этого используют prettier и/или eslint.

- https://prettier.io
- https://eslint.org

Для начала достаточно использовать prettier.

taraswww777 replied:
## Пробелы или табы?

Предпочтительнее использовать пробелы, при этом 2шт.
Пробелы одинаково отображаются в разных средах.
Но в рамках команды можно договориться о чём угодно.

taraswww777 commented on src/css/index.scss:
```diff
@@ -0,0 +1,83 @@
+}
+
+.main-title {
+    font-size: 50px;
```
предпочтительнее будет для `font-size` использовать rem вместо px

taraswww777 commented on src/css/index.scss:
```diff
@@ -0,0 +1,83 @@
+    background-size: contain;
+    transition: filter 100ms;
+    cursor: pointer;
+    &:hover {
```
### Форматирование css

Для форматирования можно вооружиться https://stylelint.io

taraswww777 reviewed (approved):
## Нет файла readme.md

В этом файле должно быть написано как поднять ваше приложение


## Нет `package-lock.json`

Нет файла `package-lock.json`, его надо добавить.

Чтобы можно при инициализации быстро установить все нужные зависимости командой `npm ci`.

taraswww777 commented:
![2024-11-04_16-12-29](https://github.com/user-attachments/assets/2602b7e8-cd28-4335-97fe-3b8be4ba0406)

Eslint сломан

Commit:
fix: удаляет Index.js

Commit:
build: рефакторинг зависимостей

- возвращает package-lock из gitignore
- фиксирует зависисмости на конкретные версии

Commit:
fix(index.ts): исправляет баги button и volume

- добавляет возможность клика по свг - иконке
- добавляет проверку на null в обработчик слайдера

Commit:
build: добавляет stylelint, правки файлов .scss

Commit:
Merge branch 'development' into add-typescript

MelWttr committed:
Merge pull request #5 from MelWttr/add-typescript

Add typescript
